pandas/tests/reshape/merge/test_merge_asof.py

```diff
@@ -1339,3 +1339,11 @@ def test_merge_index_column_tz(self):
             index=pd.Index([0, 1, 2, 3, 4]),
         )
         tm.assert_frame_equal(result, expected)
+
+
+def test_non_numerical_dtype():
+    # GH 29130
+    left = pd.DataFrame({"x": ["2019-06-01 00:09:12", "2019-06-01 00:10:29"]})
```

Comment:
Is this parametrizable over other bad inputs?

Reply:
Added a mixed DataFrame with float and string